src/LiveComponent/src/Util/LiveControllerAttributesCreator.php

```diff
@@ -80,16 +80,23 @@ public function attributesForRendering(MountedComponent $mounted, ComponentMetad
 
         $mountedAttributes = $mounted->getAttributes();
 
-        if ($isChildComponent) {
-            if (!isset($mountedAttributes->all()['data-live-id'])) {
-                $id = $deterministicId ?: $this->idCalculator
-                    ->calculateDeterministicId(key: $mounted->getInputProps()[self::KEY_PROP_NAME] ?? null);
-                $attributesCollection->setLiveId($id);
-                // we need to add this to the mounted attributes so that it is
-                // will be included in the "attributes" part of the props data.
-                $mountedAttributes = $mountedAttributes->defaults(['data-live-id' => $id]);
-            }
+        if ($mounted->hasExtraMetadata('hostTemplate') && $mounted->hasExtraMetadata('embeddedTemplateIndex')) {
+            $mountedAttributes = $mountedAttributes->defaults([
+                'data-host-template' => $mounted->getExtraMetadata('hostTemplate'),
```

Comment:
This exposes the raw Twig template name to the frontend, right? I totally get *why*, but I don't LOVE it. And I also can't think of a GREAT way around it. We could... encrypt it using the secret? Or... in theory could we, during cache warmup, use the `twig.template_iterator` service to create a "map" of "random string => template name"?

Reply:
As discussed via Slack, I wondered if this was a big deal, but as agreed that it's just to be on the safe side, I'll look into the cache warmup option.

Reply:
I've added the suggested cache warmer, and obscured the host template attribute through the resulting map.